### DISCUSSION TITLE
One builderCase - many thunks, is it achievable?

### DISCUSSION THREAD

**Discussion by msynowski** [reduxjs/redux-toolkit#1968] [Q&A] [answered]
Hi!
Is it possible to create one ```builder.addCase``` to cover multiple thunks? 
I'm refactoring an app from classic Redux to Redux Toolkit. Previously it wasn't a problem, as we were using `switch` with multiple cases over one expression.

We've got few thunks, that do pretty much the same in terms of extra reducers, but we rather keep them separated, as they cover different scenarios and got different sets of props.

Here's what it looks like right now:
```js
// register and registerExternal are just createAsyncThunk with Axios  
builder.addCase(register.pending, (state, action) => {
  state.fetchingData = true;
});

builder.addCase(registerExternal.pending, (state, action) => {
  state.fetchingData = true;
});
```
And here's my imagination of what it could look like
```js
builder.addCase([register.pending, registerExternal.pending], (state, action) => {
  state.fetchingData = true;
});
```
Is something like this achievable in Redux Toolkit?

**Comment by phryneas** [comment, marked as answer]
Yes, with `addMatcher` and [isAnyOf](https://redux-toolkit.js.org/api/matching-utilities#isanyof)

```js
builder.addMatcher(isAnyOf(register.pending, registerExternal.pending), (state, action) => {
  state.fetchingData = true;
});
```
